halogen/types.py

```diff
@@ -274,22 +305,61 @@ def deserialize(self, value, **kwargs):
             raise ValueError(u"'{amount}' has more than 2 decimal places.".format(amount=amount))
 
         value = self.amount_class(currency=currency, amount=amount)
-        return super(Amount, self).deserialize(value)
+        return super().deserialize(value)
 
 
 class Nullable(Type):
     """Nullable type."""
 
-    def __init__(self, nested_type, *args, **kwargs):
+    def __init__(self, nested_type: Type, *args, **kwargs):
```

Comment:
if I remember correctly: `nested_type` is either `Type`, or `type[Type]`, also it can be a `Schema`.

Reply:
✅